src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const resultState = state;
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(resultState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const n of action.keysToRemove) {
+          delete resultState[n];
+        }
+        break;
+
+      case 'clear':
+        Object.keys(resultState).forEach(key => delete resultState[key]);
+        break;
+    }
+  }
+
```

Comment:
`default` block in `switch` statement is required, so you should add it. It can prevent some errors in the future